packages/cactus-plugin-ledger-connector-ethereum/src/main/typescript/web-services/invoke-raw-web3eth-contract-v1-endpoint.ts

```diff
@@ -88,9 +89,44 @@ export class InvokeRawWeb3EthContractEndpoint implements IWebServiceEndpoint {
     const reqTag = `${this.getVerbLowerCase()} - ${this.getPath()}`;
     this.log.debug(reqTag);
 
+    const userInput = req.body;
+    const abi = userInput.abi;
+
+    if (!Array.isArray(abi) || abi.length == 0) {
+      throw new Error("Invalid or missing ABI in the request");
+    }
+
+    const methodName = userInput.contractMethod;
+    const method = abi.find(
+      (item: FunctionFragment) => item.name === methodName,
+    );
+
+    //getting methodparams from request
+    const contractMethodArgs = userInput.contractMethodArgs || [];
+
+    if (!method) {
+      throw new Error("Method not found in the provided ABI");
+    }
+
+    //getting methodparams from abi
+    const functionInputs = method.inputs || [];
+
+    for (const input of functionInputs) {
+      if (
+        !Object.prototype.hasOwnProperty.call(contractMethodArgs, input.name)
+      ) {
+        throw new Error(`Missing input parameter: ${input.name}`);
+      }
+
+      // Check if the input type matches the expected type
+      const isValidType = typeof userInput[input.name] === input.type;
+      if (!isValidType) {
+        throw new Error(`Invalid data type for parameter ${input.name}`);
+      }
+    }
```

Comment:
Could you please move this logic to the `invokeRawWeb3EthContract` method of connector? This way we can benefit from it while using connector directly (i.e. without an ApiServer / not through an endpoint).

Same goes for the quorum connector

Reply:
Sure!